### PR TITLE
Add Github Authentication and Header

### PR DESCRIPTION
**Pull request by uniorunr**
issue #9
 - implemented GitHub Authentication via Firebase;
 - added Header;

The live version of App - https://rss-mentor-dashboard-2185d.firebaseapp.com/

### CODE REVIEW

File: src/components/Dashboard/parser.js [hunk 1]
```diff
@@ -37,40 +35,34 @@ function getTaskStatus(name, dataObj) {
...
+
   if (score) {
-    return '✅ ' + score;
-  } else
-  if (score && getTaskStatus(currentTaskName, dataObj) === "Checking") {
-    return '🕓 ' + score;
-  } else
-  if (!score && getTaskStatus(currentTaskName, dataObj) === "Checking") {
+    return `✅ ${score}`;
+  } if (score && getTaskStatus(currentTaskName, dataObj) === 'Checking') {
```
Comment (davojta) [inline src/components/Dashboard/parser.js]: у нас разве нету уже prettier ? чтобы форматировать нормально?
if должен быть после {

File: src/components/Header/HeaderSelect/HeaderSelect.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    };
+  }
+
+  componentDidMount = () => {
+    const { mentorsList } = this.state;
+
+    firebase.auth().onAuthStateChanged((user) => {
+      const mentorFromStorage = localStorage.getItem('currentMentor');
+
+      if (user && isMentor(user.displayName, mentorsList)) {
```
Comment (davojta) [inline src/components/Header/HeaderSelect/HeaderSelect.js]: isMentor(user.displayName, mentorsList) is better to put to separate var

File: src/firebase/firebase.js [hunk 3]
```diff
@@ -19,6 +20,40 @@ class FireBase {
...
+        throw new Error(error);
+      });
+    }
+  }
+
+  static logout() {
+    firebase.auth().signOut().then(() => {
+      window.location.reload();
+    }).catch((error) => {
+      throw new Error(error);
```
Comment (davojta) [inline src/firebase/firebase.js]: it is better to create a normal error  and add console.error('logout error', e);

**Review by davojta — CHANGES_REQUESTED** [review]
Need to adjust some code style issues

**Comment by uniorunr** [comment]
I've fixed issues

File: src/components/Dashboard/parser.js [hunk 4]
```diff
@@ -44,39 +44,51 @@ function getScore(studentName, mentor, currentTaskName, dataObj) {
 
   if (score) {
     return `✅ ${score}`;
-  } if (score && getTaskStatus(currentTaskName, dataObj) === 'Checking') {
+  }
```
Comment (davojta) [inline src/components/Dashboard/parser.js:47]: thanks!

File: src/components/Header/HeaderSelect/HeaderSelect.js [hunk 5]
```diff
@@ -38,8 +38,9 @@ class HeaderSelect extends Component {
 
     firebase.auth().onAuthStateChanged((user) => {
       const mentorFromStorage = localStorage.getItem('currentMentor');
+      const mentor = isMentor(user.displayName, mentorsList);
 
-      if (user && isMentor(user.displayName, mentorsList)) {
+      if (user && mentor) {
```
Comment (davojta) [inline src/components/Header/HeaderSelect/HeaderSelect.js:43]: +1

**Review by davojta — APPROVED** [review]
LGTM!

**Pull request merged**